ML:	http://sources.redhat.com/ml/libc-hacker/2005-09/msg00002.html

Date: Mon, 05 Sep 2005 21:07:15 +0900 (JST)
Message-Id: <20050905.210715.15267870.kkojima@rr.iij4u.or.jp>
To: libc-hacker at sources dot redhat dot com
Subject: SH: A typo in lowlevellock.S
From: Kaz Kojima <kkojima at rr dot iij4u dot or dot jp>

Hi,

The appended patch fixes a typo in a low-level lock function.  It
set the correct 3rd argument for the futex syscall in loop.  Sorry
for missing such an embarrassing bug.

Regards,
	kaz

nptl/ChangeLog:
2005-09-05  Kaz Kojima  <kkojima@rr.iij4u.or.jp>

	* sysdeps/unix/sysv/linux/sh/lowlevellock.S (__lll_mutex_lock_wait):
	Fix typo in register name.

--- glibc.old/nptl/sysdeps/unix/sysv/linux/sh/lowlevellock.S	2004-10-26 04:06:44.000000000 +0900
+++ glibc/nptl/sysdeps/unix/sysv/linux/sh/lowlevellock.S	2005-09-05 19:18:25.000000000 +0900
@@ -1,4 +1,4 @@
-/* Copyright (C) 2003, 2004 Free Software Foundation, Inc.
+/* Copyright (C) 2003, 2004, 2005 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
 
    The GNU C Library is free software; you can redistribute it and/or
@@ -51,8 +51,8 @@ __lll_mutex_lock_wait:
 	SYSCALL_INST_PAD
 
 2:
-	mov	#2, r4
-	XCHG (r4, @r8, r2)
+	mov	#2, r6
+	XCHG (r6, @r8, r2)
 	tst	r2, r2
 	bf	1b
 
Signed-off-by: Robert P. J. Day <rpjday@mindspring.com>
